Library/Homebrew/cmd/upgrade.rb

```diff
@@ -5,10 +5,17 @@ module Homebrew
   def upgrade
     Homebrew.perform_preinstall_checks
 
-    if ARGV.named.empty?
+    if ARGV.include?("--all") #|| ARGV.named.empty?
+      unless ARGV.include? "--all"
```

Comment:
This will always be false since we’re in `if ARGV.include?("--all")`:

```rb
if ARGV.include?("--all")
  unless ARGV.include? "--all"  # <-- we're here
    # …
  end
end
```